packages/block-editor/src/components/link-control/index.js

```diff
@@ -133,15 +135,20 @@ function LinkControl( {
 		withCreateSuggestion = true;
 	}
 
+	const { set: setPreference } = useDispatch( preferencesStore );
+	const setSettingsOpenWithPreference = ( isOpen ) => {
+		// Should we set both?
+		setPreference( 'core/edit-site', 'linkControlSettingsDrawer', isOpen );
+		setPreference( 'core/edit-post', 'linkControlSettingsDrawer', isOpen );
```

Comment:
I don't think you should have any reference to the various editor implementations. This is the block editor package.

Rather you can pass in a "setter" function via settings which this component can access and use.

This is similar to how we access the preference


```
select( blockEditorStore ).getSettings()
				.linkControlAdvancedSettingsPreference,

```

Reply:
✅ I've fixed this.